File: core/src/main/kotlin/com/github/servb/pph/pheroes/common/common/FractionCoeff.java
package com.github.servb.pph.pheroes.common.common;

/**
 *
 * @author SerVB
 */
public final class FractionCoeff {

    public int num, denum;

    public FractionCoeff() {
        num = -1;
        denum = -1;
    }

    public FractionCoeff(final int num, final int denum) {
        this.num = num;
        this.denum = denum;
    }

    public final boolean IsValid() {
        return num != -1 && denum != -1;
    }

    public final FractionCoeff GetNormalized() {
        if (num > denum) {
            return new FractionCoeff(num / denum, 1);
        }
        if (denum > num) {
            return new FractionCoeff(1, (denum + num - 1) / num);
        }
        return new FractionCoeff(1, 1);
    }

}

File: core/src/main/kotlin/com/github/servb/pph/pheroes/common/common/FractionCoeff.kt
package com.github.servb.pph.pheroes.common.common

data class FractionCoeff(var num: Int, var denum: Int) {
    constructor() : this(-1, -1)

    fun IsValid(): Boolean {
        return num != -1 && denum != -1
    }

    fun GetNormalized(): FractionCoeff {
        return when {
            num > denum -> FractionCoeff(num / denum, 1)
            denum > num -> FractionCoeff(1, (denum + num - 1) / num)
            else -> FractionCoeff(1, 1)
        }
    }
}
